book/src/contributing/fuzzing.md

```diff
@@ -0,0 +1,27 @@
+# Fuzzing
+
+**Logos**' has a fuzzing crate powered by [afl.rs](https://github.com/rust-fuzz/afl.rs) that
+finds panics in Logos' methods.
+
+## Usage
+
+Make sure you have `cargo-afl` installed. [See the rust-fuzz afl setup guide for installation information](https://rust-fuzz.github.io/book/afl/setup.html).
+
+All of these commands assume your current directory is in the `fuzz` folder.
+
+To build the fuzz target, run `cargo afl build`.
+
+To start the fuzzer in the `fuzz` folder, run:
+
+```sh
+cargo afl fuzz -i in -o out ../target/debug/logos-fuzz
+```
+
+To replay any crashes, run:
+
+```sh
+cargo afl run logos-fuzz < out/default/crashes/crash_file
+```
+
+If you find a meaningful crash with a vague error message, send a PR to
+help improve the developer experience of Logos.
```

Comment:
````suggestion
## Usage

First, make sure you have `cargo-afl` installed,
[see the rust-fuzz afl setup guide for installation information](https://rust-fuzz.github.io/book/afl/setup.html).

Next, change your current working directory to be the `fuzz` folder.

### Building

Before fuzzing, you need to build the target with:

```bash
cargo afl build
```

### Fuzzy testing

The recommended way the run tests is with:

```bash
cargo afl fuzz -i in -o out ../target/debug/logos-fuzz
```

Note that it may run for a (very) long time before
it encounter any bug.

## Replaying a Crash

If you happen to find a bug that crashes the program,
you can reply it with

```bash
cargo afl run logos-fuzz < out/default/crashes/crash_file
```

### Reporting a Bug

If you encounter a crash and you feel the error message
is not appropriate,
please report it by opening
[an issue](https://github.com/maciejhirsz/logos/issues/new).
Don't forget to include your crash file so we can later
reproduce it.
````